pkg/plugins/policies/meshratelimit/plugin/v1alpha1/plugin_test.go

```diff
@@ -627,73 +631,218 @@ var _ = Describe("MeshRateLimit", func() {
 		Expect(bytes).To(test_matchers.MatchGoldenYAML(path.Join("testdata", "basic_egress.golden.yaml")))
 	})
 
-	It("should generate proper Envoy config for MeshGateway Dataplanes", func() {
-		// given
-		gatewayRules := core_rules.GatewayRules{
-			ToRules: core_rules.GatewayToRules{
-				ByListener: map[core_rules.InboundListener]core_rules.Rules{
-					{Address: "192.168.0.1", Port: 8080}: {{
-						Subset: core_rules.Subset{},
-						Conf: api.Conf{
-							Local: &api.Local{
-								HTTP: &api.LocalHTTP{
-									RequestRate: &api.Rate{
-										Num:      100,
-										Interval: v1.Duration{Duration: 10 * time.Second},
+	type gatewayTestCase struct {
+		name           string
+		gatewayRoutes  []*core_mesh.MeshGatewayRouteResource
+		meshhttproutes core_rules.GatewayRules
+		meshtcproutes  core_rules.GatewayRules
+		rules          core_rules.GatewayRules
+	}
+	DescribeTable("should generate proper Envoy config for MeshGateways",
+		func(given gatewayTestCase) {
+			Expect(given.name).ToNot(BeEmpty())
+			resources := xds_context.NewResources()
+
+			gateway := &core_mesh.MeshGatewayResource{
```

Comment:
can we rewrite all of this to use YAML file as an input? I know it's already been written like this but it makes it really hard to review

Reply:
I think we could do it as a separate task. I am not fully sure if it's worth adding file policy input just for one test.